### PR TITLE
UI improvement : new set

### LINKED ISSUE #11272: sort in lighttable's top filter tool bar not visible when overing decending order
**Describe the bug/issue**

In lighttable, select descending order filter and over with mouse. The icon is then non visible.

**To Reproduce**
1. Go to 'lighttable'
2. Select decending order
3. Fly over with mouse
4. See error, no icon visible

**Expected behavior**

Icon should be visible as when the ascending order is selected.

**Screenshots**

(mouse non visible when taking screenshot)

Capture d’écran de 2022-03-05 21-19-02

### PR DESCRIPTION
**Pull request by Nilvus**
This PR improves collapsible sections. It remains though combobox, entries and checkboxes to improve. Will come.
This also fix 2 issues:
- fix #11272
- reject icon that doesn't disappear correctly when block overlays is used.

And improves menuitem to have well balanced margins.

To come:
- what remains to do on collapsible sections
- improve quick filters in top toolbar with CSS
And maybe more...

### CODE REVIEW

**Comment by TurboGit** [comment]
@Nilvus : Thanks! Do not hesitate to publish screenshot on the new stuff, it is always nice to see the work on the UI.

**Comment by Nilvus** [comment]
@dterrahe: last commit update bauhaus UI from your #11257 PR. Could you just check if it's good for you? Thanks

**Comment by dterrahe** [comment]
If you are happy, I'm happy.

**Comment by Nilvus** [comment]
Last commit change all combobox (except bauhaus ones), entries and textview. Should improve visibility and contrast.

And here it is, some screenshots:

collapsible

prefs

**Comment by Nilvus** [comment]
Last commit just optimize some CSS for some transparent background items.

@TurboGit and @aurelienpierre: making last commit, I'm so seeing starting line 1720, part named // add collapsible section for those extra options that are generally not to be used. I remember collapsible section in filmic previously but there's no more (or am I missing something). Is there some cleanup to make here, at least removing lines 1720 to 1737? Or maybe more (or less!).

**Comment by Nilvus** [comment]
And now, better quick filter toolbars with last commit. This commit so improve bauhaus-combobox and make them behave quite like classic combobox (with now hover effect too!). See:

index

Actually, there's a bug in collections modules but it's more that new CSS make visible an old issue in collections modules:

bug

And this probably need some Gtk rework and @AlicVB is working on this part.

@phweyland: I hope you will like this visual effect on quick filter toolbar (note that collect filter PR by @AlicVB will also update that part, thanks to your initial work.

**Comment by TurboGit** [comment]
> @TurboGit and @aurelienpierre: making last commit, I'm so seeing starting line 1720, part named // add collapsible section for those extra options that are generally not to be used. I remember collapsible section in filmic previously but there's no more (or am I missing something). Is there some cleanup to make here, at least removing lines 1720 to 1737? Or maybe more (or less!).

I'm not finding those lines in current darktable.css nor in the version on this commit. Any pointer to this part of code ?

**Comment by Nilvus** [comment]
> > @TurboGit and @aurelienpierre: making last commit, I'm so seeing starting line 1720, part named // add collapsible section for those extra options that are generally not to be used. I remember collapsible section in filmic previously but there's no more (or am I missing something). Is there some cleanup to make here, at least removing lines 1720 to 1737? Or maybe more (or less!).
> 
> I'm not finding those lines in current darktable.css nor in the version on this commit. Any pointer to this part of code ?

I forgot to precise that I talk about `src/iop/filmic.c` file.

I will post a new commit that simplify code for control button (arrow of collapsible sections). @TurboGit: you propose to add +/- but maybe actual arrow is also good. What do you think about keeping it? I'm ok though to change and add +/- but maybe not needed.

**Comment by TurboGit** [comment]
> I forgot to precise that I talk about `src/iop/filmic.c` file.

That's the old deprecated filmic module. So we don't want to touch it.

**Comment by Nilvus** [comment]
> > I forgot to precise that I talk about `src/iop/filmic.c` file.
> 
> That's the old deprecated filmic module. So we don't want to touch it.

Forget that. So, do I revert my little add on this file? Just note that if i revert it, control button of collapsible section could be visually broken. So if someone see this module, it's better keep my change.

**Comment by TurboGit** [comment]
> So if someone see this module, it's better keep my change.

Agreed, we do not want to rework deprecated module but this little change to make it visually ok is fine by me.

**Review by TurboGit — CHANGES_REQUESTED** [review]
Something seems broken with the dark theme. The contrast is very high for the collapsible sections:

image

and

image

As you can see also the collapsible section header is slightly smaller than the area. This is true in all themes BTW.

**Comment by TurboGit** [comment]
Also the text inside the filters (like `nom de fichier`) is not centered.

image

**Comment by Nilvus** [comment]
Strange, that's not what I have on my side. I would not let something like that but maybe I made a mistake when pushing here... Will check

**Comment by TurboGit** [comment]
> Strange, that's not what I have on my side. I would not let something like that but maybe I made a mistake when pushing here... Will check

I think alike, so I have double checked on my side that I didn't have some local changes... Really strange !

**Comment by Nilvus** [comment]
> Also the text inside the filters (like `nom de fichier`) is not centered.
> 
> image

This is an already known issue and not related to that PR. I already pointed that in rework by @phweyland (and was also not related to his PR). It's something related to bauhaus-combobox. And bauhaus file seems a nightmare for some expert developer on darktable, so it's not for me to update that. Padding and margin do not seems to have any real effect on those bauhaus (I hate them)

**Comment by Nilvus** [comment]
Oh, note that this alignement issue was less visible before, when combobox were not really visible (with same background that around). If you want to go inside bauhaus.c code and fix that, I'm all for! ;-)

**Comment by Nilvus** [comment]
> > Strange, that's not what I have on my side. I would not let something like that but maybe I made a mistake when pushing here... Will check
> 
> I think alike, so I have double checked on my side that I didn't have some local changes... Really strange !

I understand, i have read darker but it's dark theme. I didn't have test it (most of the time when it's ok in darker, it's ok in dark). And error in header is visible, it's quite less visible in darker and grey theme so I miss it, especially in my 4k screen!

**Comment by AlicVB** [comment]
@Nilvus @TurboGit : for the sort combobox : 
it's a combobox, so the width of the widget is the width of the largest possible text. So we will have margins somewhere, even with "regular" combobox.
With that in mind what would you like to have as alignment ? left is what "regular" combobox do (and it's just a parameter to set for bauhaus combobox) imho center will looks strange imho (and not in line with what we have elsewhere in the UI)

I ask that because I just start the sort integration into my filtering lib, so I can change what you want (well almost...) !

**Comment by Nilvus** [comment]
@TurboGit: last commit fix issues (header and dark theme)

**Comment by Nilvus** [comment]
> @Nilvus @TurboGit : for the sort combobox : it's a combobox, so the width of the widget is the width of the largest possible text. So we will have margins somewhere, even with "regular" combobox. With that in mind what would you like to have as alignment ? left is what "regular" combobox do (and it's just a parameter to set for bauhaus combobox) imho center will looks strange imho (and not in line with what we have elsewhere in the UI)
> 
> I ask that because I just start the sort integration into my filtering lib, so I can change what you want (well almost...) !

Some margins work but not to correct the vertical alignment issue (not completely at least). Apart of that, it's good. I don't care if the text is aligned at right or left here. It's actually good as it is for me, except the vertical alignment pointed by @TurboGit.

**Comment by Nilvus** [comment]
@TurboGit: Wait before merging, I see an issue in some menuitem (in metadata preset for example in import module) and I need to check if it's from my PR or not. And how to fix it.

**Comment by AlicVB** [comment]
@Nilvus : sorry, I've read too quickly and thought the problem was horizontal alignment... For vertical, indeed, it would require change inside bauhaus code itself :(

**Comment by Nilvus** [comment]
Ok, last commit fix 2 issues I've seen:

- first about menuitem. My initial commit was to remove too big margin in most menuitem (especially those opened when clicking on bottom left bar in darkroom (presets/styles). This is more a workaround as those menuitem have by default 0 margin and padding but have a big left margin anyway (probably something hardcoded! My initial commit though break menuitem seen in import module (metadata presets) as it is better written (no margin/padding in Gtk code, so all set with CSS only. So margin-left with negative thing is a workaround. But I don't know where and what to change to have all menuitem by default with no margin without CSS. So this would need some rework.
And so, I hope I don't forget some menuitem to check and not broken anything (merging in master would be a good test).
- Adding new background to bauhaus combobox is not wanted by default when they are in a popover (like those on icons in bottom right bar in darkroom), so I added a line in transparent options.

**Comment by Nilvus** [comment]
@TurboGit: ready so for review now.

And as a remember, you probably missed my question about expander icon: _you propose to add +/- but maybe actual arrow is also good. What do you think about keeping it? I'm ok though to change and add +/- but maybe not needed._

**Comment by TurboGit** [comment]
> And as a reminder, you probably missed my question about expander icon: _you propose to add +/- but maybe actual arrow is also good. What do you think about keeping it? I'm ok though to change and add +/- but maybe not needed._

Yes, missed it! And yes, the current arrow icon is ok for me now. We will revisit this later after more user and field testing if we have reports. Thanks for the hard work.

**Comment by Nilvus** [comment]
I will after that PR post an issue if someone want to update menuitem to avoid such CSS workaround. It's not a good thing that different menuitem start with different margins (none or some) hardcoded...

**Comment by TurboGit** [comment]
@Nilvus : I have also an improvement to propose. The collapsible section header is very (too) similar in color to the actual collapsible section. It is a bit difficult to see the section header. Take for example the `color calibration` module where we have two collapsible sections next to each other. When both are expanded it is difficult to see the separation between both. Can this be improved?

See in darker and dark theme:

image

image

**Comment by TurboGit** [comment]
We have in current master:

image

And the combo-boxes in the section are flat and this is cleaner as it doesn't "break" the collapsible section.

**Comment by AlicVB** [comment]
@Nilvus : in @TurboGit screenshots, I really the two arrow (the for collapsible section and the one for combobox really confusing... What do you think about : 
1. change the dropdown icon in bauhaus combobox with something similar than the one in regular combobox
2. implement the +- icon like what we speak earlier in the thread...

Eventually, both can be done... I can help you for the bauhauscombobox if you want : just ask :)

**Comment by Nilvus** [comment]
> @Nilvus : in @TurboGit screenshots, I really find the two arrow (the for collapsible section and the one for combobox really confusing... What do you think about :
> 
>     1. change the dropdown icon in bauhaus combobox with something similar than the one in regular combobox
> 
>     2. implement the +- icon like what we speak earlier in the thread...
> 
> 
> Eventually, both can be done... I can help you for the bauhauscombobox if you want : just ask :)

I would here be more for similar than in regular combobox. More consistent. I also see that the little border I set to combobox (regular and bauhaus) is not applied to bauhaus. Or to be more precise, it is applied but not visible. So probably something remains hardcoded in Gtk code that override my CSS setting. If you see quickly how to remove border overriding with Gtk code, I would love it.

In @TurboGit screenshot, we don't correctly see bauhaus-combobox as set in my PR here. See what I have on my side by now (without thin border set, it's not really good visuallyl)

image

@TurboGit: in this screenshot (and last commit), I update colors to answer your last comment. To improve visibility, I also add a top and bottom margin on collapsible sections.

**Comment by Nilvus** [comment]
Last commit finish combobox improvements. Code changed in bauhaus.c is done by @AlicVB. Thanks to him again for his help. With that last update, bauhaus combobox are same visual as classic combobox and now hover effect on all things (and even arrow, changed to be consistent too with classic combobox). This also allow CSS border to be applied in bauhaus-combobox.

I let only combobox in modules (like export or color calibration ones) as they are by default. I test with same combobox update proposed in other places (make combobox more visible). On those modules with multiples combobox, this is really too much and much better visually as it is already.

@TurboGit: except if you see some errors, it's all now good for me. I also simplify a little code in paint.c

image

image

**Review by TurboGit — APPROVED** [review]
Looks good now, thanks!

File: data/themes/darktable.css [hunk 1]
```diff
@@ -2078,9 +2035,9 @@ treeview#delete-dialog:selected
 }
 
 /* Sliders states */
-#bauhaus-popup:hover,
-#bauhaus-combobox:hover,
-#bauhaus-slider:hover
+#bauhaus-popup *:hover,
+#bauhaus-combobox *:hover,
+#bauhaus-slider *:hover
```
Comment (dterrahe) [inline data/themes/darktable.css:2040]: @Nilvus @TurboGit this broke slider hovering prelight. Bauhaus widgets are not containers, so they don't have children that you can refer to with *

For combobox there is now a `#bauhaus-combobox:hover` above that gets picked so would be a duplicate here.

Comment (TurboGit) [reply data/themes/darktable.css:2040]: Thanks for noticing, I'll let @Nilvus fix that.

Comment (Nilvus) [reply data/themes/darktable.css:2040]: Oh shit, just forgotten them. This was test when I didn't have what I wanted. As you can see, I put bauhaus-combo for example some lines above. Thanks for noticing that. Will fix it. Only bauhaus-combobox have children (bauhaus-popup especially).

**Pull request merged**